atlasdb-console/build.gradle

```diff
@@ -30,14 +30,18 @@ dependencies {
     testImplementation 'org.junit.jupiter:junit-jupiter'
     testImplementation project(':atlasdb-client')
 
-    testImplementation('org.jmock:jmock-legacy') {
-        exclude group: 'org.hamcrest'
-    }
     testImplementation('org.jmock:jmock') {
       exclude group: 'org.hamcrest'
     }
+    testImplementation('org.jmock:jmock-imposters') {
+        exclude group: 'org.hamcrest'
+    }
     testImplementation 'org.gmock:gmock'
 }
 
 tasks.checkImplicitDependenciesMain.enabled = false
 tasks.checkUnusedDependenciesMain.enabled = false
+
+test {
+    jvmArgs '--add-opens=java.base/java.lang=ALL-UNNAMED'
+}
```

Comment:
This solves tests under `com.palantir.atlasdb.console.groovy`

Reply:
Not so nice solution. Basically, JDK17 has stricter encapsulation levels for base packages. This is a work around. We are using gmock for our groovy tests and the latest update is on 2013 :(

Reply:
I see. This makes sense, but probably best to include context (comment?)